aws-iot-custommetric/src/main/java/com/amazonaws/iot/custommetric/CreateHandler.java

```diff
@@ -48,6 +49,9 @@ public CreateHandler() {
             createCustomMetricResponse = proxy.injectCredentialsAndInvokeV2(
                     createRequest, iotClient::createCustomMetric);
         } catch (Exception e) {
+            if (e instanceof ResourceAlreadyExistsException) {
+                throw new CfnAlreadyExistsException(e);
+            }
             return Translator.translateExceptionToProgressEvent(model, e, logger);
         }
```

Comment:
Why not add a separate catch block for this exception? Also, should we add some logging?

Once you add a separate catch block for RAEE. Add another catch block for RuntimExceptions, we shouldn't be catching "Exception" as is, if there are any checked exceptions being thrown then we need to handle them instead of wrapping everything under the same blanked

Reply:
Changed as requested